playbooks/maas-pre-flight.yml

```diff
@@ -12,30 +12,40 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-
-- name: Check for RHOSP
-  hosts: localhost
-  gather_facts: false
+- name: Setup local facts
+  hosts: all
+  gather_facts: true
   become: true
   tasks:
-    - name: Check for /etc/rhosp-release
-      stat:
-        path: /etc/rhosp-release
-      register: osp_release_file
+    - name: Ensure local facts directory exists
+      file:
+        dest: "/etc/ansible/facts.d"
+        state: directory
+        group: "root"
+        owner: "root"
+        mode:  "0755"
+        recurse: no
 
-    - name: Check for /usr/bin/tripleo-ansible-inventory
-      stat:
-        path: /usr/bin/tripleo-ansible-inventory
-      register: tripleo_inventory_file
+    - name: Remove local maas.facts
+      file:
+        dest: "/etc/ansible/facts.d/maas.fact"
+        state: absent
 
-    - name: Set deploy_osp fact
-      set_fact:
-        deploy_osp: "{{ (osp_release_file.stat.exists and tripleo_inventory_file.stat.exists) | bool }}"
```

Comment:
This is actually needed because the OSP vars task littered through the plays explicitly look for the fact over the variable with the dynamic generation changes. This was done to begin the work to remove the need for the OSP vars to be included in every play.

Reply:
I understood,  but deploy_osp isn't working consistently throughout my testing so I moved deploy_osp into rpcr_dynamic inventory, thus this is no longer needed.  Technically rpcr_dynamic inventory should be able to figure out all OSP vars in the future.
I did put rhosp release logic later in preflight code and that part is needed.
IMO using set fact is not as direct and not working well in my tests so I had to move it away.